h2/src/main/scala/org/scalarelational/h2/H2Datastore.scala

```diff
@@ -18,40 +16,31 @@ import org.scalarelational.table.Table
 abstract class H2Datastore private() extends SQLDatastore with Logging {
   protected def this(mode: H2ConnectionMode = H2Memory(org.powerscala.Unique()),
                      username: String = "sa",
-                     password: String = "sa") = {
+                     password: String = "sa") {
     this()
     dbUsername := username
     dbPassword := password
     modeProperty := mode
   }
 
-  protected def this(dataSource: DataSource) = {
```

Comment:
It is moderately redundant, but I think I want to try to make `dataSource` immutable in the future so it will need to be supported. Also, just because it's not being used in ScalaRelational doesn't mean someone utilizing the framework wouldn't prefer to do it this way.